ompi/mca/coll/base/coll_base_reduce.c

```diff
@@ -583,6 +583,7 @@ int ompi_coll_base_reduce_intra_in_order_binary( const void *sendbuf, void *recv
                                           op, io_root, comm, module,
                                           data->cached_in_order_bintree,
                                           segcount, max_outstanding_reqs );
+    free(tmpbuf_free);
```

Comment:
Not good, if `io_root == rank` then the `tmpbuf` is used via `use_this_recvbuf` in the send operation.

Reply:
Is the expectation here that since ompi_coll_base_reduce_generic uses 'use_this_recvbuf', that it will free it if the function call is successful, and that it only needs to be freed here if that function fails, which makes this Coverity report a false positive?

Reply:
Never mind, i see the use of use_this_recvbuf after this. So I think the existing tmpbuf_free needs to be left where it was, and I need a free above only if ret != MPI_SUCCES

Reply:
@bosilca I just pushed an update that I think addresses the concern here. Take a look at the new change and let me know what you think.

Reply:
@bosilca ping